File: docs/code-quality/codesnippet/CPP/ca2208-instantiate-argument-exceptions-correctly_1.cpp
using namespace System;
 
namespace Samples1 
{    
	public ref class Book    
	{
     private: initonly String^ _Title;
 
    public:
        Book(String^ title)        
		{            
			// Violates this rule (constructor arguments are switched)            
			if (title == nullptr)                
				throw gcnew ArgumentNullException("title cannot be a null reference (Nothing in Visual Basic)", "title");
 
            _Title = title;        
		}
 
        property String^ Title        
		{            
			String^ get()            
			{                
				return _Title;            
			}        
		}    
	};
}

File: docs/code-quality/codesnippet/CSharp/ca2208-instantiate-argument-exceptions-correctly_1.cs
using System;
 
namespace Samples1
{    
    public class Book    
    {        
        private readonly string _Title;
 
        public Book(string title)        
        {            
            // Violates this rule (constructor arguments are switched)            
            if (title == null)                
                throw new ArgumentNullException("title cannot be a null reference (Nothing in Visual Basic)", "title");
 
            _Title = title;        
        }
 
        public string Title        
        {            
            get { return _Title; }        
        }
    }
}